components/amazon_selling_partner/amazon_selling_partner.app.mjs

```diff
@@ -1,11 +1,168 @@
+        } = payload;
+        return {
+          options: orders?.map(({ AmazonOrderId }) => ({
+            label: AmazonOrderId,
+            value: AmazonOrderId,
+          })) || [],
+          context: {
+            next,
+          },
+        };
+      },
+    },
+    status: {
+      type: "string[]",
+      label: "Status",
+      description: "Filter inbound shipments by status",
+      options: [
+        "WORKING",
+        "SHIPPED",
+        "RECEIVING",
+        "CANCELLED",
+        "DELETED",
+        "CLOSED",
+        "ERROR",
+        "IN_TRANSIT",
+        "DELIVERED",
+        "CHECKED_IN",
+      ],
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUrl() {
+      return "https://sellingpartnerapi-na.amazon.com";
+    },
+    _makeRequest({
+      $ = this, path, ...opts
+    }) {
+      return axios($, {
+        url: `${this._baseUrl()}${path}`,
+        headers: {
+          "x-amz-access-token": this.$auth.oauth_access_token,
+        },
+        ...opts,
+      });
+    },
+    getOrder({
+      orderId, ...opts
+    }) {
+      return this._makeRequest({
+        path: `/orders/${orderId}`,
+        ...opts,
+      });
+    },
+    getOrderMetrics(opts = {}) {
+      return this._makeRequest({
+        path: "/sales/v1/orderMetrics",
+        ...opts,
+      });
+    },
+    getInventorySummaries(opts = {}) {
+      return this._makeRequest({
+        path: "/fba/inventory/v1/summaries",
+        ...opts,
+      });
+    },
+    listMarketplaces(opts = {}) {
+      return this._makeRequest({
+        path: "/sellers/v1/marketplaceParticipations",
+        ...opts,
+      });
+    },
+    listOrders(opts = {}) {
+      return this._makeRequest({
+        path: "/orders/v0/orders",
+        ...opts,
+      });
+    },
+    listInboundShipments(opts = {}) {
+      return this._makeRequest({
+        path: "/fba/inbound/v0/shipments",
+        ...opts,
+      });
+    },
+    listReports(opts = {}) {
+      return this._makeRequest({
+        path: "/reports/2021-06-30/reports",
+        ...opts,
+      });
+    },
+    listProductPricing(opts = {}) {
+      return this._makeRequest({
+        path: "/products/pricing/v0/competitivePrice",
+        ...opts,
+      });
+    },
+    async *paginate({
+      fn, params, resourceKey, hasPayload = true, max,
+    }) {
+      let hasMore, count = 0;
+      do {
+        const response = await fn({
+          params,
+        });
+        const payload = hasPayload
+          ? response.payload
+          : response;
+        const items = payload[resourceKey] || [];
+        for (const item of items) {
+          yield item;
+          if (max && ++count >= max) {
+            return;
+          }
+        }
+        hasMore = false;
+        if (payload?.NextToken) {
+          params.NextToken = payload.NextToken;
+          hasMore = true;
+        }
```

Comment:
This block of code seems duplicated

Reply:
It looks duplicated, but it's not. :) Some endpoints use `NextToken` (capitalized), and some use `nextToken` (not capitalized), so I wanted to allow for either one.
```
        hasMore = false;
        if (payload?.NextToken) {
          params.NextToken = payload.NextToken;
          hasMore = true;
        }
        if (payload?.nextToken) {
          params.nextToken = payload.nextToken;
          hasMore = true;
        }
```

Reply:
Ah I see the casing difference - makes sense!